### ISSUE TITLE
Jest encountered an unexpected token with react + konva and/or react-konva

### ISSUE THREAD

**Issue by damiangreen** [konvajs/react-konva#608]
I'm getting the same error as posted on StackOverflow:

https://stackoverflow.com/questions/67919198/jest-encountered-an-unexpected-token-with-react-konva-and-or-react-konva

I've also tried the comment in the suggestion `yarn add konva canvas` to no effect.

This seems to be prevneting us from upgrading

**Comment by lavrton** [comment]
Please make a very small repository with such a problem to reproduce.

**Comment by maxxxymum** [comment]
Faced the same issue with Jest, react-testing-library.

**Comment by maxxxymum** [comment]
@damiangreen   I found a solution to fix this issue:

1. Install `canvas` https://www.npmjs.com/package/canvas. `npm i canvas`.
2. Update your configs as mentioned in this [comment](https://github.com/facebook/jest/issues/11753#issuecomment-912532776).

**Comment by damiangreen** [comment]
thanks @maxxxymum  I'll give it a try

**Comment by damiangreen** [comment]
Then i receive `Cannot find module 'canvas' from 'node_modules/konva/lib/index-node.js'`
I tried following the steps on https://github.com/konvajs/konva#4-nodejs-env
with no luck

**Comment by OliverBrace1** [comment]
If you have anything new on this that would be wonderful as I'm starting to think the way I resolved this issue is directly causing https://github.com/konvajs/react-konva/issues/630

**Comment by lavrton** [comment]
The issue should be resolved with `konva@8.3.6`.

Also you may need this config for jest:

```js
"transformIgnorePatterns": [
  "node_modules/(?!(konva)).*\\.js$"
]
```

If you still have an issue, create a small repo to reproduce the issue.

**Comment by Amoodaa** [comment]
@lavrton I made a repro repo, this is a fresh create-react-app with typescript template
https://github.com/Amoodaa/konva-jest-repro
